knpu/get-authenticated-user.md

```diff
@@ -9,60 +9,90 @@ create the products in `FeatureContext`.
-class - near the other product function. These two functions will be similar, so
-we should reuse the logic. Copy the internals of `thereAreProducts`, make a new
-`private function createProducts`. Pass it `$count` as an argument and also an optional
-User object which will be the author for those products. Now, add an if statement
-that says, if `$author` is passed then, `$product->setAuthor()`. I already have that
-relationship setup with in Doctrine. Great!
+```bash
+$ ./vendor/bin/behat features/web/product_admin.feature:13
+```
 
-In `thereAreProducts`, change the body of this function to `$this->createProducts($count);`.
-Do the same thing in `iAuthorProducts` for now. Clearly, this is still not setting
-the author. But I want to see if it executes first and then we'll worry about setting
-the author.
+Great - copy the `iAuthorProducts()` function code and paste it into our handy `FeatureContext`
+class - near the other product function:
 
+[[[ code('a2d4b5fdea') ]]]
+
+These two functions will be similar, so we should reuse the logic. Copy the internals
+of `thereAreProducts`, make a new `private function createProducts()`. Pass it `$count`
+as an argument and also an optional User object which will be the author for those products:
+
+[[[ code('8bbd967c08') ]]]
+
+Now, add an if statement that says, if `$author` is passed then, `$product->setAuthor()`:
+
+[[[ code('954b443f9e') ]]]
+
+I already have that relationship setup with in Doctrine. Great!
+
+In `thereAreProducts()`, change the body of this function to `$this->createProducts($count);`:
+
+[[[ code('95e36de658') ]]]
+
+Do the same thing in `iAuthorProducts()` for now:
+
+[[[ code('acd4d6cbd4') ]]]
+
+Clearly, this is still not setting the author. But I want to see if it executes first
+and then we'll worry about setting the author.
 
 ## Who is "I" in a Scenario?
 
 Cool! It runs... and fails because anonymous *is* still shown on the page. The question
 now is: how do we get the current user? The step says "I author". But who is "I" in this
-case? In `product_admin.feature`, you can see that "I" is whomever we logged in as.
-We didn't specify what the username should be for that user, but whoever is logged
-in is who "I" represents.
+case? In `product_admin.feature`:
+
+[[[ code('88c42ae944') ]]]
+
+You can see that "I" is whomever we logged in as. We didn't specify what the username
+should be for that user, but whoever is logged in is who "I" represents.
 
 When we worked with the `ls` scenarios earlier, we needed to share the command output
 string between the steps of a scenario. In this case, we have a similar need: we
 need to share the user object from the step where we log in, with the step where
-"I" author some products. To share data between steps, create a new `private $currentUser;`.
+"I" author some products. To share data between steps, create a new `private $currentUser;`:
+
+[[[ code('100d047bcd') ]]]
```

Comment:
@weaverryan Should we add more context here and wrap this property with class in the code block? I show a single line of property declaration here in code block now